src/snowflake/connector/config_manager.py

```diff
@@ -334,44 +362,48 @@ def add_subparser(self, new_child: ConfigManager) -> None:
             new_child: The ConfigManager to be nested under the current one.
         Notes:
             We currently don't support re-nesting a ConfigManager. Only nest a
-            parser under another one once.
+            manager under another one once.
         """
         self._check_child_conflict(new_child.name)
-        self._sub_parsers[new_child.name] = new_child
+        self._sub_managers[new_child.name] = new_child
 
         def _root_setter_helper(node: ConfigManager):
             # Deal with ConfigManagers
             node._root_manager = self._root_manager
             node._nest_path = self._nest_path + node._nest_path
-            for sub_parser in node._sub_parsers.values():
-                _root_setter_helper(sub_parser)
+            for sub_manager in node._sub_managers.values():
+                _root_setter_helper(sub_manager)
             # Deal with ConfigOptions
             for option in node._options.values():
                 option._root_manager = self._root_manager
                 option._nest_path = self._nest_path + option._nest_path
 
         _root_setter_helper(new_child)
 
+    # Alias for the old name of add_submanager in the first release, please use the
+    #  new name, as this might get deprecated in the future.
+    add_subparser = add_submanager
```

Comment:
same as above, I think we can log a warning

Reply:
See https://github.com/snowflakedb/snowflake-connector-python/pull/1680#discussion_r1310936729